### PR TITLE
Expand tilde when calculating path

### PR DESCRIPTION
**Pull request by KiruyaMomochi**
Freedesktop's fontconfig replaces leading `~` in the path with the user's home directory.

This config makes our path calculating behavior the same as theirs.

### CODE REVIEW

**Comment by KiruyaMomochi** [comment]
If we are only a [parser](https://en.wikipedia.org/wiki/Parsing), we should do no more than *syntax analysis*, or *constructing a parse tree*. Any resolving work, like what this PR suggests, should not happen. That's what `parse_config` and `read_to_string` does.

But our current implementation does more. We resolve elements like "include", which recursively finds and merge all the config files. Such resolution must require some kind of path calculation, like convert `prefix="xdg"` to some real file path. 

Then to what extent of *resolution* should we provide? This PR resolves all `~` into `$HOME`, but there are other possibilities. We can also resolve lazily only when we need to include configs.

File: src/types/dir.rs [hunk 1]
```diff
@@ -80,6 +96,13 @@ macro_rules! define_calculate_path {
                         std::env::var($xdg_env).unwrap_or_else(|_| $xdg_fallback.into()),
                     )
                     .join(self.path.as_str()),
+                };
+
+                if path.starts_with("~") {
+                    let home = config_home().unwrap_or("/".to_string());
+                    std::path::Path::new(&home).join(path.strip_prefix("~").unwrap())
+                } else {
+                    path
```
Comment (Riey) [inline src/types/dir.rs]:
```suggestion
                if let Some(stripped_path) = path.strip_prefix("~") {
                    let home = config_home().unwrap_or("/".to_string());
                    std::path::Path::new(&home).join(stripped_path)
                } else {
                    path
```

**Comment by Riey** [comment]
> If we are only a [parser](https://en.wikipedia.org/wiki/Parsing), we should do no more than _syntax analysis_, or _constructing a parse tree_. Any resolving work, like what this PR suggests, should not happen. That's what `parse_config` and `read_to_string` does.
> 
> But our current implementation does more. We resolve elements like "include", which recursively finds and merge all the config files. Such resolution must require some kind of path calculation, like convert `prefix="xdg"` to some real file path.
> 
> Then to what extent of _resolution_ should we provide? This PR resolves all `~` into `$HOME`, but there are other possibilities. We can also resolve lazily only when we need to include configs.

We export raw config parsing option via [parse_config_parts](https://docs.rs/fontconfig-parser/latest/fontconfig_parser/fn.parse_config_parts.html), `caculate_path` function is just helper function which doesn't affect to any field of `Dir`.

`FontConfig` does some useful processings such as recursive include so I think it's fine to do more stuffs while doesn't touching raw [ConfigPart](https://docs.rs/fontconfig-parser/latest/fontconfig_parser/enum.ConfigPart.html) fields.

**Comment by KiruyaMomochi** [comment]
Don't know why it's closed...

**Review by Riey — APPROVED** [review]

**Pull request merged**